File: medplum-java/medplum-model/src/main/java/com/fhir/MolecularSequence_Quality.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Integer;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableMolecularSequence_Quality.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface MolecularSequence_Quality {
  Optional<CodeableConcept> standardSequence();

  Optional<decimal> truthTP();

  Optional<decimal> queryFP();

  Optional<Quantity> score();

  Optional<decimal> precision();

  Optional<String> id();

  Optional<Molecularsequence_qualityType> type();

  Optional<decimal> fScore();

  Optional<decimal> truthFN();

  Optional<decimal> recall();

  Optional<MolecularSequence_Roc> roc();

  Optional<decimal> gtFP();

  Optional<CodeableConcept> method();

  Optional<List<Extension>> modifierExtension();

  Optional<decimal> queryTP();

  Optional<Integer> end();

  Optional<Integer> start();

  Optional<List<Extension>> extension();

  static ImmutableMolecularSequence_Quality.Builder builder() {
    return ImmutableMolecularSequence_Quality.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/MolecularSequence_Quality.ts
import { CodeableConcept } from "./CodeableConcept";
import { decimal } from "./decimal";
import { Extension } from "./Extension";
import { Molecularsequence_qualityType } from "./Molecularsequence_qualityType";
import { MolecularSequence_Roc } from "./MolecularSequence_Roc";
import { Quantity } from "./Quantity";

export interface MolecularSequence_Quality {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    type: Molecularsequence_qualityType | null | undefined;
    standardSequence: CodeableConcept | null | undefined;
    start: number | null | undefined;
    end: number | null | undefined;
    score: Quantity | null | undefined;
    method: CodeableConcept | null | undefined;
    truthTP: decimal | null | undefined;
    queryTP: decimal | null | undefined;
    truthFN: decimal | null | undefined;
    queryFP: decimal | null | undefined;
    gtFP: decimal | null | undefined;
    precision: decimal | null | undefined;
    recall: decimal | null | undefined;
    fScore: decimal | null | undefined;
    roc: MolecularSequence_Roc | null | undefined;
}
